private/bufpkg/bufimage/build_image.go

```diff
@@ -137,10 +142,7 @@ func getBuildResult(
 	if err != nil {
 		if err == reporter.ErrInvalidSource {
 			if len(errorsWithPos) == 0 {
-				return newBuildResult(
-					nil,
-					nil,
-					nil,
+				return newFailedBuildResult(
```

Comment:
I introduced this new helper/constructor to aid readability and lessen the `nil` verbosity in this function.